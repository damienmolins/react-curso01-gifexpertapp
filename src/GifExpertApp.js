import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';


const GifExpertApp = ({ defaultCategories = [] }) => {

    //const [categories, setCategories] = useState(['One Punch']);
    const [categories, setCategories] = useState(defaultCategories);

    return <>
        <h1>GifExpertApp</h1>
        <hr />
        <AddCategory setCategories={setCategories} />
        <ol>
            {
                categories.map(
                    category => (
                        <GifGrid
                            key={category}
                            category={category}
                        />
                    )
                )
            }
        </ol>
    </>;
}

GifExpertApp.propTypes = {
    defaultCategories: PropTypes.array
}

export default GifExpertApp;