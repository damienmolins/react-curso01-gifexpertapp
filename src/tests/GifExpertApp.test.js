import React from 'react';
import { shallow } from 'enzyme';
import GifExpertApp from '../GifExpertApp';

describe('pruebas para <GifExpertApp />', () => {

    test('snapshot', () => {
        const wrapper = shallow(<GifExpertApp />);
        expect(wrapper).toMatchSnapshot();
    })

    test('debe de mostrar una lista de categorias', () => {
        const defaultCategories = ['Dragon Ball', 'One Punch'];
        const wrapper = shallow(<GifExpertApp defaultCategories={defaultCategories} />)
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(defaultCategories.length);
    })

})
