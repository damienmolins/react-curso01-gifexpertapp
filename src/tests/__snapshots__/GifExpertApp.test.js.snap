// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas para <GifExpertApp /> debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h1>
    GifExpertApp
  </h1>
  <hr />
  <AddCategory
    setCategories={[Function]}
  />
  <ol>
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
  </ol>
</Fragment>
`;

exports[`pruebas para <GifExpertApp /> snapshot 1`] = `
<Fragment>
  <h1>
    GifExpertApp
  </h1>
  <hr />
  <AddCategory
    setCategories={[Function]}
  />
  <ol />
</Fragment>
`;
